src/QueryEditor.tsx

```diff
@@ -84,10 +86,10 @@ export const QueryEditor: ComponentType<Props> = ({ datasource, onChange, onRunQ
         <div className="gf-form-label">
           <Label>Additional JSON Data</Label>
         </div>
-        <div className="gf-form">
+        <div className="gf-form grafana-json-datasource-editor">
           <CodeEditor
-            width="500px"
-            height="100px"
+            width="100%"
+            height="175px"
```

Comment:
Shouldn't (or could not) height also be dynamic?

Reply:
Dynamic would be certainly nice

Reply:
This was my initial idea aswell, but this didn't seem so simple to achieve.

I did investigate this a bit, this is the code in grafana/ui for the codeEditor which also shows the div thats wrapping the editor: https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/components/Monaco/CodeEditor.tsx#L92-L104

which led me to the MonacoEditor:
https://github.com/react-monaco-editor/react-monaco-editor/blob/master/src/editor.tsx

and then ended up on this open issue: https://github.com/microsoft/monaco-editor/issues/794#issuecomment-583367666
All of the suggestions propose to use `editor.onDidChangeModelDecorations`, but we can't access that when using the grafana/ui CodeEditor.

So imo. we would need a pr to grafana/ui CodeEditor first to add one of the workarounds from the issue above, ie. when `height: auto` is set or by adding a new property to toggle it. Without this i don't think that we can have a auto growing code editor.

But i can give it a try and see if i can apply the workaround locally in node_modules for a local build. But the PR to grafana + waiting for the next release will take a while - so this could be a intermediate solution.